dames.c

```diff
@@ -1,6 +1,246 @@
 #include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <stdbool.h>
 
-int dummy_fun(void){
-	printf("Hello yall\n");
-	return 0;
+#define GRIDSIZE 10
+
+#define WHITE_CHECKER 1
+#define WHITE_DAME 2
+#define BLACK_CHECKER -1
+#define BLACK_DAME -2
+#define CASE_VIDE 0
+#define IA 1
+#define HUMAIN 0
+
+
+typedef struct Case {
+    int line;
+    int col;
+} Case;
+
+typedef struct checkboard{
+    int grid[GRIDSIZE][GRIDSIZE];
+    int nb_white, nb_black;
+}checkboard;
+
+void show_line(int i, int checkboard[GRIDSIZE][GRIDSIZE]){
+    int j;
+    printf(" %2d", i);
+    for (j=0; j< 10; j++){
+        switch(checkboard[i][j]){
+            case WHITE_CHECKER : 
+                printf("| W ");
+                break;
+            case BLACK_CHECKER :
+                printf("| B ");
+                break;
+            /*case WHITE_DAME :
+                printf("| [W] ");
+                break;
+            case BLACK_DAME :
+                printf("| [W] ");
+                break;*/
+            default :
+                printf("| - ");
+                break;
+        }
+    }
+    printf("|\n");
+}
+
+void show_board(int checkboard[GRIDSIZE][GRIDSIZE]){
+    int i;
+    printf("     0   1   2   3   4   5   6   7   8   9 \n");
+    printf("   ----------------------------------------- \n");
+    for (i=0; i < GRIDSIZE; i++){
+        show_line (i, checkboard);
+    }
+    printf("   ----------------------------------------- \n");
+    printf("     0   1   2   3   4   5   6   7   8   9 \n");
+}
+
+void init_checkboard(checkboard* game){
+    
+	game->nb_white = 20;
+    game->nb_black = 20;
+
+    int i, j;
+
+    for(i=0; i<GRIDSIZE; i++){
+        for (j=0; j<GRIDSIZE; j++){
+            game->grid[i][j] = 0;
+        }
+    }
+
+    for(i=0; i<4; i++){
+        // lorsque pair mettre des pions noirs
+        if(i%2 == 0){j =1;}
+        else {j = 0;}
+
+        for (j=j; j<GRIDSIZE; j+=2){
+            game->grid[i][j] = BLACK_CHECKER;
+        }
+    }
+
+    for(i=6; i<10; i++){
+        // Lorque pair mettre des pions blancs
+        if(i%2 == 0){j =1;}
+        else {j = 0;}
```

Comment:
That's not a "normal" indentation

```
if(cond){

}
else {

}
```

or 
```
if()
{
}
else
{
}
```

are the only two possibles options. 

You might be tempted to do one liners such as:

`j = (i%2) ? 1  : 0;`

but that's generally a bad idea too. Because that's not very much readable.